consent/manager_sql.go

```diff
@@ -364,8 +364,35 @@ WHERE
 		return nil, sqlcon.HandleError(err)
 	}
 
+	return m.resolveHandledConsentRequests(a)
+}
+
+func (m *SQLManager) FindPreviouslyGrantedConsentRequestsByUser(subject string, limit, offset int) ([]HandledConsentRequest, error) {
+	var a []sqlHandledConsentRequest
+
+	if err := m.db.Select(&a, m.db.Rebind(`SELECT h.* FROM
+	hydra_oauth2_consent_request_handled as h
+JOIN
+	hydra_oauth2_consent_request as r ON (h.challenge = r.challenge)
+WHERE
+		r.subject=? AND r.skip=FALSE
+	AND
+		(h.error='{}' AND h.remember=TRUE)
+LIMIT ? OFFSET ?
+`), subject, limit, offset); err != nil {
+		if err == sql.ErrNoRows {
+			return nil, errors.WithStack(errNoPreviousConsentFound)
```

Comment:
So one issue with this is that the HTTP handler function will return `errNoPreviousConsentFound` which is a "dumb" error (meaning it will return a 500 status code with no context in the API). On the same hand, this error is required to be returned for the `consent_strategy` to work. Since you duplicated the methods and this particular method is not used in the `consent_strategy`, you can remove the `err == sql.ErrNoRows` check and just return `return nil, sqlcon.HandleError(err)`.

Please also add a test for this. It should check that `sqlcon.ErrNoRows` is returned when we expect that no session exist in the store. This will return a proper 404 error in the API which I think is to be expected here.

In the handler, you may want to choose to ignore the error of type `sqlcon.ErrNoRows` and instead just write an empty array. This will return `[]` instead of a 404 which might be a better suited response.

Reply:
Thanks for your help and your feedback! I have gladly implemented your suggestions. 

However, I'm a little unsure at this point and need your help: Does `db.Select` really return an `ErrNoRows` error if no rows could be found? According to my understanding of SQLX / Go SQL documentation, this does not happen here. I have not been able to force such behaviour in my experiments.

Or did I misunderstand and the manager should generate and return an `sqlcon.ErrNoRows`?